elastic/kibana
# Add the query delay mechanism to the inventory rule type

Closes #170531

## Summary

This PR adds the query delay mechanism to the inventory rule type.

![image](https://github.com/elastic/kibana/assets/12370520/c511fd13-f297-4d1e-8c60-7cd10c23621f)

This PR also changes the inventory rule type consumer from `infrastructure` to `observability`:

<img src="https://github.com/elastic/kibana/assets/12370520/e93da707-0e4e-4e49-80eb-17c4aebbb188" width=400 />


## 🧪 How to test
- On serverless
    - Create an inventory rule and make sure, 
        - by default, the query execution has 15s delay
        - consumer field in alert is `observability`
    - Change the query settings delay in the Observability rule setting and ensure the specified delay is applied to the related query
    
    ![image](https://github.com/elastic/kibana/assets/12370520/b6acc154-0f15-40c0-912a-8b8c2235a253)

- On stateful
    - This mechanism is not applied to stateful, so the query execution time should not have any delay
    - Make sure consumer for inventory rule is `infrastructure` as before
